Label-Microservice/notebooks/embeddings.py

```diff
@@ -0,0 +1,137 @@
+from bs4 import BeautifulSoup
+import requests
+from fastai.core import parallel, partial
+from collections import Counter
+from tqdm import tqdm
+import torch
+import pandas as pd
+from inference import InferenceWrapper
+from pathlib import Path
+from urllib import request as request_url
+import logging
+
+
+def find_max_issue_num(owner, repo):
+    """
+    Find the maximum issue number associated with a repo.
+
+    Returns
+    -------
+    int
+        the highest issue number associated with this repo.
+    """
+    url = f'https://github.com/{owner}/{repo}/issues'
+    r = requests.get(url)
+    if not r.ok:
+        r.raise_for_status()
+    soup = BeautifulSoup(r.content, 'html.parser')
+    # get grey text under issue preview cards
+    issue_meta = soup.find('span', class_="opened-by").text
+    # parse the first issue number visible, which is also the highest issue number
+    issue_num = issue_meta.strip().split('\n')[0][1:]
+    return int(issue_num)
+
+def get_issue_text(num, idx, owner, repo, skip_issue=True):
+    """
+    Get the raw text of an issue body and label.
+
+    Returns
+    ------
+    dict
+        {'title':str, 'body':str}
+    """
+    url = f'https://github.com/{owner}/{repo}/issues/{num}'
+    if requests.head(url).status_code != 200:
+        if skip_issue:
+            return None
+        raise Exception(f'Status code is not 200:\n{url} is not an issue.')
```

Comment:
Can you append the status code and detailed response message in the exception? It will help you debug. e.g. the request can be rate limited.

Reply:
Done